core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionSupervisor.scala

```diff
@@ -26,56 +28,101 @@ import scala.util.control.NonFatal
-  // It is only written by this actor.
+  /* A note on metrics:
+   *  We instrument four things here:
+   *  1. The number of submissions in each state across the whole system.
+   *  2. The number of workflows in each state across the whole system.
+   *  3. The number of submissions in each state, by workspace, for any workspace with at least one running submission.
+   *  4. The number of workflows in each state, by workspace and submission, for any submission with at least one running workflow.
+   *
+   *  Aggregating over #3 and #4 will give you strange results, as metrics will disappear when submissions or workspaces finish running.
+   *  #3 and #4 are meant to be used as a way to diagnose the behaviour of a particular workspace or submission.
+   *
+   *  Every SubmissionMonitorActor calculates #3 and sends it in the SaveCurrentWorkflowStatusCounts message. Therefore two submissions
+   *  running in the same workspace will update activeSubmissionStatusCounts to the same value once per poll interval. This is silly but fine.
+   *
+   *  The fields below are marked volatile because they are read by a separate statsd thread.
+   *  They are only written by this actor.
+   */
   @volatile
-  private var currentWorkflowStatusCounts: Map[WorkflowStatus, Int] = Map.empty
+  private var activeSubmissionStatusCounts: Map[WorkspaceName, Map[SubmissionStatus, Int]] = Map.empty
   @volatile
-  private var currentSubmissionStatusCounts: Map[SubmissionStatus, Int] = Map.empty
+  private var activeWorkflowStatusCounts: Map[UUID, Map[WorkflowStatus, Int]] = Map.empty
+  @volatile
+  private var globalSubmissionStatusCounts: Map[SubmissionStatus, Int] = Map.empty
+  @volatile
+  private var globalWorkflowStatusCounts: Map[WorkflowStatus, Int] = Map.empty
+
+  override def preStart(): Unit = {
+    super.preStart()
+
+    registerGlobalJobExecGauges()
+    system.scheduler.schedule(0 seconds, submissionPollInterval, self, RefreshGlobalJobExecGauges)
+  }
 
   override def receive = {
-    case SubmissionStarted(workspaceContext, submissionId, credential) =>
-      val child = startSubmissionMonitor(workspaceContext, submissionId, credential)
+    case SubmissionStarted(workspaceName, submissionId) =>
+      val child = startSubmissionMonitor(workspaceName, submissionId, bucketCredential)
       scheduleNextCheckCurrentWorkflowStatus(child)
-      initGauge(workspaceContext, submissionId)
+      registerDetailedJobExecGauges(workspaceName, submissionId)
 
-    case SaveCurrentWorkflowStatusCounts(workflowStatusCounts, submissionStatusCounts, reschedule) =>
-      this.currentWorkflowStatusCounts = workflowStatusCounts
-      this.currentSubmissionStatusCounts = submissionStatusCounts
+    case SaveCurrentWorkflowStatusCounts(workspaceName, submissionId, workflowStatusCounts, submissionStatusCounts, reschedule) =>
+      this.activeWorkflowStatusCounts += submissionId -> workflowStatusCounts
```

Comment:
`+=` is correct here (I think) -- we want to _replace_ the entry for the given workspace/submission with the current counts.